src/compiler/parser.ts

```diff
@@ -1124,6 +1124,14 @@ namespace ts {
             return token === t && tryParse(nextTokenCanFollowModifier);
         }
 
+        function nextTokenIsOnSameLineAndCanFollowModifier() {

```

Comment:
But why do some people needs checks, and others do not?  That's the part I find confusing.  How would I know from my code?